src/compiler/parser.ts

```diff
@@ -3976,8 +4092,10 @@ module ts {
             parseExpected(SyntaxKind.FunctionKeyword);
             node.asteriskToken = parseOptionalToken(SyntaxKind.AsteriskToken);
             node.name = node.flags & NodeFlags.Default ? parseOptionalIdentifier() : parseIdentifier();
-            fillSignature(SyntaxKind.ColonToken, /*yieldAndGeneratorParameterContext:*/ !!node.asteriskToken, /*requireCompleteParameterList:*/ false, node);
-            node.body = parseFunctionBlockOrSemicolon(!!node.asteriskToken, Diagnostics.or_expected);
+            let isGenerator = node.asteriskToken != undefined;

```

Comment:
That's fine. I usually reserve `!=` and `==` for null tests, as it covers both `null` and `undefined` without needing two checks.